contracts/stork_pyth_adapter/contracts/StorkPythAdapter.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity >=0.8.24 <0.9.0;
+
+import "@pythnetwork/pyth-sdk-solidity/IPyth.sol";
+import "@pythnetwork/pyth-sdk-solidity/PythStructs.sol";
+
+/**
+ * @title A port of the IPyth interface that supports Stork price feeds
+ */
+contract StorkPythAdapter is IPyth {
+    IStork public stork;
+    int32 private exponent = -18;
+    uint64 private confidenceInterval = 0;
+
+    constructor(address _stork) {
+        stork = IStork(_stork);
+    }
+
+    function convertInt192ToInt64(int192 value) public pure returns (int64) {
+        require(value >= type(int64).min && value <= type(int64).max, "Overflow");
```

Comment:
need to convert prices to int64 to conform to their interface, which could lead to overflow